### PR TITLE
sys-power/throttled: new package, add 0.10.0

### PR DESCRIPTION
**Pull request by vimproved**
I know this was discussed in IRC that this package has been requested before, so here it is. Custom systemd and openrc services are necessary because the ones from upstream hardcode a venv in /opt.

### CODE REVIEW

**Comment by dlan17** [comment]
I think proxy-maint team are quite conservative to accept new packages unless you do have good reasons (anyone can confirm)? - like a) it's important package that used by wide users b) other package already in portage tree that depend on it ..

if answers is NO, you probably could submit it to GURU overlay more easily [1]

[1] https://wiki.gentoo.org/wiki/Project:GURU

File: sys-power/throttled/files/throttled [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/sbin/openrc-run
+command="/usr/bin/throttled"
+pidfile="/var/run/throttled.pid"
```
Comment (dlan17) [inline sys-power/throttled/files/throttled:3]: using /run would be just fine, as /var/run is symbol link to /run

also, I'd suggest follow the convention to rename the file to throttled.initd to make it explicitly as openrc init script..

File: sys-power/throttled/throttled-0.10.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{10..12} )
+inherit python-single-r1 systemd
+
+DESCRIPTION="Workaround for Intel throttling issues in Linux."
+HOMEPAGE="https://github.com/erpalma/throttled"
+SRC_URI="https://github.com/erpalma/throttled/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+
```
Comment (dlan17) [inline sys-power/throttled/throttled-0.10.0.ebuild:12]: I'd suggest renaming the source tarball to ${P}.gh.tar.gz to give a hint that it's retrieved from github.com (also a convention here & it's only my suggestion)

File: sys-power/throttled/throttled-0.10.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	${PYTHON_DEPS}
+	$(python_gen_cond_dep '
+		dev-python/dbus-python[${PYTHON_USEDEP}]
+		dev-python/pygobject[${PYTHON_USEDEP}]
+	')
+	sys-apps/pciutils
+"
+
+src_install() {
+	python_newscript throttled.py throttled
```
Comment (dlan17) [inline sys-power/throttled/throttled-0.10.0.ebuild:28]: I'd personally like to keep 'default' which would help to handle docs at least..

src_install() {
default
..
}

**Comment by vimproved** [comment]
> I think proxy-maint team are quite conservative to accept new packages unless you do have good reasons (anyone can confirm)? - like a) it's important package that used by wide users b) other package already in portage tree that depend on it ..
> 
> if answers is NO, you probably could submit it to GURU overlay more easily [1]
> 
> [1] https://wiki.gentoo.org/wiki/Project:GURU

I'm aware, but apparently this package has been pretty widely requested according to @thesamesam.
```
sam_>  oh feel free to pr that to ::gento obtw
sam_>  a bunch of people have asked for it
```

**Comment by dlan17** [comment]
> > I think proxy-maint team are quite conservative to accept new packages unless you do have good reasons (anyone can confirm)? - like a) it's important package that used by wide users b) other package already in portage tree that depend on it ..
> > if answers is NO, you probably could submit it to GURU overlay more easily [1]
> > [1] https://wiki.gentoo.org/wiki/Project:GURU
> 
> I'm aware, but apparently this package has been pretty widely requested according to @thesamesam. If other members of proxy-maint think this should go to GURU instead I'll send it there.
> 
> ```
> vimproved> thermald conflicts with throttled
> vimproved>  https://github.com/erpalma/throttled
> sam_>  oh feel free to pr that to ::gento obtw
> sam_>  a bunch of people have asked for it
> ```

Ok, fine by me, given I'm the maintainer of thermald, I probably could also help to proxy maintain this package too..

File: sys-power/throttled/files/throttled (hunk 1 above)
Comment (dlan17) [reply sys-power/throttled/files/throttled]: file throttled.initd?

File: sys-power/throttled/files/throttled.service [hunk 4]
```diff
@@ -0,0 +1,11 @@
+[Unit]
+Description=Stop Intel throttling
```
Comment (dlan17) [inline sys-power/throttled/files/throttled.service:2]: white space in the end, please drop

File: sys-power/throttled/throttled-0.10.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+inherit python-single-r1 systemd
+
+DESCRIPTION="Workaround for Intel throttling issues in Linux."
+HOMEPAGE="https://github.com/erpalma/throttled"
+SRC_URI="https://github.com/erpalma/throttled/archive/refs/tags/v${PV}.tar.gz -> ${P}.gh.tar.gz"
+
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64"
```
Comment (dlan17) [inline sys-power/throttled/throttled-0.10.0.ebuild:16]: I think ~x86 should also work? but I'm fine if leave it alone (if not tested, and probably not many people using)

File: sys-power/throttled/throttled-0.10.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,36 @@
+# Copyright 2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{10..12} )
+inherit python-single-r1 systemd
+
+DESCRIPTION="Workaround for Intel throttling issues in Linux."
```
Comment (dlan17) [inline sys-power/throttled/throttled-0.10.0.ebuild:9]: I know the description is copied from original website, but it sounds a little bit weird to me.. if you can't think better improvement, then I'd be just fine ..

**Comment by dlan17** [comment]
https://github.com/erpalma/throttled#thermald

can you check about this?
it describe a conflict, I think it's a run-time configuration issue? It's not strictly that throttled & thermald should block each other? but maybe have ewarn in pkg_postinst() to give user some hints?

**Comment by dlan17** [comment]
== quote from README

Note that some kernels (e.g. [linux-hardened](https://www.archlinux.org/packages/extra/x86_64/linux-hardened/)) will prevent from writing to /dev/mem too. Specifically, you need a kernel with CONFIG_DEVMEM and CONFIG_X86_MSR set.

so I think it's a good idea to check CONFIG_DEVMEM and CONFIG_X86_MSR

use eclass linux-info, you can consult the ebuild in erpalma overlay [1] for example

[1] https://github.com/erpalma/throttled-overlay/blob/master/sys-power/throttled/throttled-0.9.2.ebuild

File: sys-power/throttled/throttled-0.10.0.ebuild (hunk 5 above)
Comment (thesamesam) [reply sys-power/throttled/throttled-0.10.0.ebuild:16]: I prefer not to add it blindly unless someone tested it in a chroot at least, although even better on real x86 given this is hw sensitive

File: sys-power/throttled/throttled-0.10.0.ebuild (hunk 6 above)
Comment (thesamesam) [reply sys-power/throttled/throttled-0.10.0.ebuild:9]: Let's drop the full stop at the end either way, but if we can come up with something more descriptive, great

File: sys-power/throttled/throttled-0.10.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+HOMEPAGE="https://github.com/erpalma/throttled"
+SRC_URI="https://github.com/erpalma/throttled/archive/refs/tags/v${PV}.tar.gz -> ${P}.gh.tar.gz"
+
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
```
Comment (thesamesam) [inline sys-power/throttled/throttled-0.10.0.ebuild:13]: pop REQUIRED_USE under KEYWORDS (following ~skel.ebuild order)

File: sys-power/throttled/throttled-0.10.0.ebuild (hunk 5 above)
Comment (vimproved) [reply sys-power/throttled/throttled-0.10.0.ebuild:15]: I'm pretty sure all the laptop models that this package is applicable for are amd64 anyway, see the list in https://github.com/erpalma/throttled

**Comment by vimproved** [comment]
I'm just going to tag @erpalma here to make sure they're aware of this PR since they maintain an upstream overlay

**Comment by dlan17** [comment]
hi @vimproved, I've pushed this package with linux-info checking added, and other minor changes..

hi @erpalma, you can update README to let users know that throttled ebuild is already in official portage tree, and since you are the upstream maintainer, feel free to co-maintain this package, just let us know or send a PR to change metadata.xml

thanks you all!

**Comment by vimproved** [comment]
> hi @vimproved, I've pushed this package with linux-info checking added, and other minor changes..
> 
> hi @erpalma, you can update README to let users know that throttled ebuild is already in official portage tree, and since you are the upstream maintainer, feel free to co-maintain this package, just let us know or send a PR to change metadata.xml
> 
> thanks you all!

Thank you!

**Comment by erpalma** [comment]
Hi guys, sorry for my inaction lately, I've been super busy @work. It's amazing that the ebuild is now in official tree! Thank you all!

**Pull request closed without merging**